covid_tracking/app.py

```diff
@@ -76,8 +76,27 @@ def add_percapita_stats(df):
             df[c + '_pc'] = 1E6 * df[c] / df['population']
     return df
 
+def add_derived_metrics(df,endpoint):
+    """Adding metrics derived from the base stats:
+        newCases: new cases on this day for this state
+        posPerTest: positive test results/ total tests
+    :param df: dataFrame
+    """
+    df['posPerTest'] = df['positive']/df['total']
```

Comment:
Can you check that those fields (positive and total) are first present since it's applied to all endpoint fetches?  Minor point, but would be nice to have that safety in case they change a field name in the API.

Reply:
Great points! Thanks for the feedback— please let me know if there are other areas I should address. Happy to contribute :)